BridgeAppSDK/SBAProfileItem.swift

```diff
@@ -0,0 +1,400 @@
+//
+//  SBAProfileItem.swift
+//  BridgeAppSDK
+//
+//  Copyright © 2017 Sage Bionetworks. All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without modification,
+// are permitted provided that the following conditions are met:
+//
+// 1.  Redistributions of source code must retain the above copyright notice, this
+// list of conditions and the following disclaimer.
+//
+// 2.  Redistributions in binary form must reproduce the above copyright notice,
+// this list of conditions and the following disclaimer in the documentation and/or
+// other materials provided with the distribution.
+//
+// 3.  Neither the name of the copyright holder(s) nor the names of any contributors
+// may be used to endorse or promote products derived from this software without
+// specific prior written permission. No license is granted to the trademarks of
+// the copyright holders even if such marks are included in this software.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+
+import Foundation
+import ResearchUXFactory
+
+@objc
+public protocol SBAProfileItem: NSObjectProtocol {
+    /**
+     key is used to access a specific profile item, and so must be unique across all SBAProfileItems
+     within an app.
+     */
+    var key: String { get }
+    
+    /**
+     sourceKey is the profile item's key within its internal data storage. By default it will be the
+     same as key, but can be different if needed (e.g. two different profile items happen to map to
+     the same key in different storage types).
+     */
+    var sourceKey: String { get }
+    
+    /**
+     demographicKey is the profile item's key in the demographic data upload schema. By default it will
+     be the same as key, but can be different if needed.
+     */
+    var demographicKey: String { get }
+    
+    /**
+     itemType specifies what type to store the profileItem as.
+     */
+    var itemType: String { get }
```

Comment:
Could you use an enum instead? Map the enum to a string? You can use `NS_ENUM_STRING` in an obj-c file to define string identifiers. Or is this the class type?